app/services/optout/OptOutService.scala

```diff
@@ -73,28 +76,45 @@ class OptOutService @Inject()(itsaStatusUpdateConnector: ITSAStatusUpdateConnect
                                                          hc: HeaderCarrier,
                                                          ec: ExecutionContext): Future[Map[TaxYear, ITSAStatus]] =
-    itsaStatusService.getStatusTillAvailableFutureYears(previousYear).map(_.view.mapValues(_.status).toMap.withDefaultValue(ITSAStatus.NoStatus))
-
-  def makeOptOutUpdateRequest()(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[ITSAStatusUpdateResponse] = {
+    itsaStatusService
+      .getStatusTillAvailableFutureYears(previousYear)
+      .map(_.view.mapValues(_.status)
+        .toMap
+        .withDefaultValue(ITSAStatus.NoStatus)
+      )
+
+  def makeOptOutUpdateRequest()(implicit user: MtdItUser[_],
+                                hc: HeaderCarrier,
+                                ec: ExecutionContext): Future[ITSAStatusUpdateResponse] = {
     recallOptOutProposition().flatMap { proposition =>
-      proposition.optOutPropositionType.map {
-        case _: OneYearOptOutProposition =>
-          makeOptOutUpdateRequest(proposition, proposition.availableTaxYearsForOptOut.head)
-        case _: MultiYearOptOutProposition =>
-          OptionT(repository.fetchSavedIntent())
-            .map(intentTaxYear => makeOptOutUpdateRequest(proposition, intentTaxYear))
-            .flatMap(responsesFuture => OptionT.liftF(responsesFuture))
-            .getOrElse(ITSAStatusUpdateResponseFailure.defaultFailure())
-      } getOrElse Future.successful(ITSAStatusUpdateResponseFailure.defaultFailure())
+      proposition
+        .optOutPropositionType
+        .map {
+          case _: OneYearOptOutProposition =>
+            makeOptOutUpdateRequest(proposition, proposition.availableTaxYearsForOptOut.head)
+          case _: MultiYearOptOutProposition =>
+            OptionT(repository.fetchSavedIntent())
+              .map(intentTaxYear => makeOptOutUpdateRequest(proposition, intentTaxYear))
+              .flatMap(responsesFuture => OptionT.liftF(responsesFuture))
+              .getOrElse(ITSAStatusUpdateResponseFailure.defaultFailure())
+        }.getOrElse {
+          Future(ITSAStatusUpdateResponseFailure.defaultFailure())
+        }
     }
   }
 
   def makeOptOutUpdateRequest(optOutProposition: OptOutProposition, intentTaxYear: TaxYear)
                              (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[ITSAStatusUpdateResponse] = {
+
     val yearsToUpdate = optOutProposition.optOutYearsToUpdate(intentTaxYear)
     val responsesSeqOfFutures = makeUpdateCalls(yearsToUpdate)
-    Future.sequence(responsesSeqOfFutures).
-      map(responsesSeq => findAnyFailOrFirstSuccess(responsesSeq))
+    Future.sequence(responsesSeqOfFutures)
+      .map(responsesSeq => findAnyFailOrFirstSuccess(responsesSeq))
+      .map { res =>
+        val auditModel = OptOutAuditModel.generateOptOutAudit(optOutProposition, intentTaxYear, res)
+        auditingService.extendedAudit(auditModel)
```

Comment:
I noticed that for the majority of audit calls in the repo that they were done in the controller. Is there a reason that this was done in the service instead?

Reply:
Sure that's fine then! Just was curious